### PR TITLE
Deliver Microsoft.Net.CoreRuntime.targets in the UwpCoreRuntimeSdk package

### PR DESCRIPTION
**Pull request by zamont**
Changes include:
- Specify paths to appx files that are delivered in the RID-specific packages
- Move Microsoft.Net.CoreRuntime.targets into repo and consume new appx file paths in Microsoft.Net.CoreRuntime.targets
- Move Microsoft.Build.Net.CoreRuntimeTask.dll source into repo and build in binaries build. This is used by Microsoft.Net.CoreRuntime.targets. Only WireUpCoreRuntimeTask was edited from before and that's to accommodate the new paths to the appx files in the UwpCoreRuntimeSdk NuGet package.

### CODE REVIEW

**Comment by zamont** [comment]
The resulting packages look like this:

# Microsoft.Net.UWPCoreRuntimeSdk
```
\build\Microsoft.Net.UWPCoreRuntimeSdk.props (consumed automatically by VS, points to CoreRuntime.targets)
\tools\CoreRuntime
    \Microsoft.Build.Net.CoreRuntimeTask.dll (contains MSBuild tasks used by CoreRuntime.targets)
    \Microsoft.Cci.dll (used by MSBuild tasks)
    \Microsoft.Net.CoreRuntime.targets
    \Microsoft.Net.CoreRuntime.settings.targets

No runtime.json.
NuGet dependencies on runtime packages -- required for .props / .targets to be automatically imported
```

# runtime.win10-{arch}-Microsoft.Net.UWPCoreRuntimeSdk
```
\build\runtime.win10-{arch}-Microsoft.Net.UWPCoreRuntimeSdk.props (consumed automatically by VS, provides properties)
\build\runtime.win10-{arch}-Microsoft.Net.UWPCoreRuntimeSdk.targets (consumed automatically by VS, provides path to appx package)
\tools
    \AppxLocal\uwpshim.exe
    \Appx\Microsoft.NET.CoreRuntime.2.1.appx
    \CoreRuntime\copywin32resources.exe
```

**Comment by zamont** [comment]
PTAL @weshaggard 
cc: @joperezr @joshfree 

I have tested these changes by privately building the UWP metapackage and taking a dependency on it in VS, and CoreCLR on UWP compilation works fine. However, we are still blocked by dotnet/core-eng#1094. The CoreRuntime appx package is not signed so F5 launch fails.

**Comment by zamont** [comment]
cc: @nattress

**Comment by zamont** [comment]
It looks like Wes is out of office right now, and he's the primary person who's been approving my changes to the package. @joshfree do you know who should take a look at this?

**Comment by joshfree** [comment]
@zamont talk to @joperezr and/or @ericstj

**Comment by zamont** [comment]
Jose and/or Eric St. John, can you please take a look? I'm hoping to merge this in the next hour.

File: src/pkg/projects/Microsoft.Net.UWPCoreRuntimeSdk/Microsoft.Net.UWPCoreRuntimeSdk.pkgproj [hunk 1]
```diff
@@ -15,13 +21,25 @@
...
   </ItemGroup>
 
+  <!-- Reference runtime packages directly so that MSBuild .props / .targets get imported correctly in VS" -->
+  <Target Name="InjectNuGetDependencyOnRuntimePackages"
+          BeforeTargets="GenerateNuSpec"
+          DependsOnTargets="DetermineRuntimeDependencies"
+          Condition="'$(PackageTargetRuntime)' == ''">
+    <ItemGroup>
+      <Dependency Include="@(RuntimeDependency)">
+        <TargetFramework>uap10.0.15138</TargetFramework>
```
Comment (joperezr) [inline src/pkg/projects/Microsoft.Net.UWPCoreRuntimeSdk/Microsoft.Net.UWPCoreRuntimeSdk.pkgproj]: Don't use this directly, if you rebase your changes with master, you'll see that there is a property with this value now. You should use `$(UAPvNextTFM)` instead.

File: src/pkg/projects/Microsoft.Net.UWPCoreRuntimeSdk/contents/identity/Microsoft.Net.CoreRuntime.targets [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+          <_FullAppLibrariesUnfiltered Include="@(PackagingOutputs)" Condition="'%(Extension)' == '.dll'" />
+        </ItemGroup>
+
+        <ConvertToAbsolutePath Paths="$(IntermediateOutputPath)">
+          <Output TaskParameter="AbsolutePaths" PropertyName="ResWOutputFullPath"/>
+        </ConvertToAbsolutePath>
+
+        <PropertyGroup Condition="'$(SkipMergingFrameworkResources)' == ''">
+          <!--
+            When building ProjectN apps in retail mode we need to skip including the framework resources in the app because we optimize the exception stack
```
Comment (joperezr) [inline src/pkg/projects/Microsoft.Net.UWPCoreRuntimeSdk/contents/identity/Microsoft.Net.CoreRuntime.targets]: Not sure if it's right to have branch names like `ProjectN` here in a comment on a target that will be imported by all UWP projects.

File: src/uwp/build.proj [hunk 3]
```diff
@@ -34,4 +34,8 @@
     <Copy SourceFiles="@(CMakeOutput)"
           DestinationFolder="$(UWPOutputDir)"/>
   </Target>
+
+  <Target Name="BuildMSBuildTask" DependsOnTargets="GetLatestCommitHash">
+    <MSBuild Projects="$(MSBuildThisFileDirectory)Microsoft.Build.Net.CoreRuntimeTask/Microsoft.Build.Net.CoreRuntimeTask.builds" />
```
Comment (joperezr) [inline src/uwp/build.proj]: Given that it is a .builds file, should we build this node in parallel? (pass in `/m`)

File: src/uwp/Microsoft.Build.Net.CoreRuntimeTask/Microsoft.Build.Net.CoreRuntimeTask.csproj [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    </Compile>
+    <Compile Include="WireUpCoreRuntime.cs" />
+    <Compile Include="MergePriConfigTask.cs" />
+    <Compile Include="MergeResWFilesTask.cs" />
+    <Compile Include="ResourceHandlingTask.cs" />
+  </ItemGroup>  
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Cci">
+      <Version>4.0.0-rc4-24217-00</Version>
```
Comment (joperezr) [inline src/uwp/Microsoft.Build.Net.CoreRuntimeTask/Microsoft.Build.Net.CoreRuntimeTask.csproj]: Just random question, any reason why not reference a stable version of the packge?

File: src/pkg/projects/Microsoft.Net.UWPCoreRuntimeSdk/contents/identity/Microsoft.Net.CoreRuntime.targets (hunk 2 above)
Comment (zamont) [inline src/pkg/projects/Microsoft.Net.UWPCoreRuntimeSdk/contents/identity/Microsoft.Net.CoreRuntime.targets]: Not that this justifies the usage, but I'm now crawling the Microsoft.NetNative.targets (the VS integration for .NET Native toolchain) and I see `ProjectN` referenced everywhere. If you would prefer, I can clear this up, but the usage on line 190 where it is a property name must remain unchanged.

File: src/uwp/Microsoft.Build.Net.CoreRuntimeTask/Microsoft.Build.Net.CoreRuntimeTask.csproj (hunk 4 above)
Comment (zamont) [inline src/uwp/Microsoft.Build.Net.CoreRuntimeTask/Microsoft.Build.Net.CoreRuntimeTask.csproj]: I don't see any stable Cci version, so I went with the latest that I saw

**Comment by zamont** [comment]
@joperezr, this UWPCoreRuntimeSdk package is essentially packaging open CoreCLR for consumption by UWP projects in VS, instead of packaging it in an MSI from a closed branch from before. We previously packaged all of the binaries into this NuGet package but this change provides the MSBuild targets so that VS F5 works correctly. Hope that provides additional context.

File: src/pkg/projects/Microsoft.Net.UWPCoreRuntimeSdk/Microsoft.Net.UWPCoreRuntimeSdk.pkgproj (hunk 1 above)
Comment (zamont) [inline src/pkg/projects/Microsoft.Net.UWPCoreRuntimeSdk/Microsoft.Net.UWPCoreRuntimeSdk.pkgproj]: Will address in following commit...

File: src/uwp/build.proj (hunk 3 above)
Comment (zamont) [inline src/uwp/build.proj]: I don't see /m being used anywhere in this repo, so I'm not too convinced we need it here.

**Comment by zamont** [comment]
I plan to merge this when CI is green, barring any additional feedback.

**Pull request merged**